stdlib/float.mli

```diff
@@ -601,6 +601,21 @@ module Array : sig
       @raise Invalid_argument if the length of [l] is greater than
       [Sys.max_floatarray_length].*)
 
+  (** {1:comparison Comparison} *)
+
+  val equal : (float -> float -> bool) -> t -> t -> bool
+  (** [equal eq a b] is [true] if and only if [a] and [b] have the
+      same length [n] and for all [i] in \[[0];[n-1]\], [eq a.(i) b.(i)]
+      is [true].
+
+      @since 5.4 *)
+
+  val compare : (float -> float -> int) -> t -> t -> int
+  (** [compare cmp a b] orders [a] and [b] in lexicographic order
+      (see {!List.compare}) using [cmp] to compare elements.
```

Comment:
Is the `List.compare` reference still a good one now that we compare the length before comparing the elements?

Reply:
I had the reference for the explanation of lexicographic order but yes I dropped it since that docstring also mentions that `cmp` is called even if the lists have different lengths.